cpp/include/cudf/detail/utilities/algorithm.cuh

```diff
@@ -25,4 +29,47 @@ __device__ __forceinline__ T accumulate(Iterator first, Iterator last, T init, B
   }
   return init;
 }
+
+/**
+ * @brief Utility for calling thrust::copy_if.
+ *
+ * Workaround for thrust::copy_if bug (https://github.com/NVIDIA/thrust/issues/1302)
+ * where it cannot iterate over int-max values `distance(first,last) > int-max`
+ * This calls thrust::copy_if in 2B chunks instead.
+ */
+template <typename InputIterator,
+          typename StencilIterator,
+          typename OutputIterator,
+          typename Predicate>
+OutputIterator copy_if(rmm::exec_policy policy,
```

Comment:
I would prefer passing a stream view instead of an execution policy to align with other libcudf APIs.

Reply:
The caller will look like this:
```
cudf::detail::copy_if(rmm::execution_policy(stream),....);
```

This is similar to what we see if using `thrust::copy_if`:
```
thrust::copy_if(rmm::execution_policy(stream),....);
```

Thus, passing `execution_policy` is necessary here.

Reply:
> passing execution_policy is necessary here

Passing a stream would work as well, right? e.g. 
```cpp
cudf::detail::copy_if(..., rmm::cuda_stream_view stream) {
  ...
  thrust::copy_if(rmm::execution_policy(stream.value()),....);
  ...
}
```
My point is that we should consistently expose streams as opposed to execution policies for libcudf APIs. The latter is thrust-centric and there is no such concept in libcudf yet.

Reply:
Okay that makes sense.

Reply:
Done.